src/net.cpp

```diff
@@ -686,12 +688,18 @@ bool CNode::ReceiveMsgBytes(Span<const uint8_t> msg_bytes, bool& complete, mapMs
             i->second += msg.m_raw_message_size;
 
             // Update the output parameter that holds received bytes per message type
-            auto j = msgtype_bytes.find(msg.m_type);
-            if (j == msgtype_bytes.end()) {
-                j = msgtype_bytes.find(NET_MESSAGE_TYPE_OTHER);
+            auto j = msgtype_countbytes.find(msg.m_type);
+            if (j == msgtype_countbytes.end()) {
+                j = msgtype_countbytes.find(NET_MESSAGE_TYPE_OTHER);
             }
-            assert(j != msgtype_bytes.end());
-            j->second += msg.m_raw_message_size;
+            assert(j != msgtype_countbytes.end());
+
+            // I'm guessing the C++ here could use some improvement..
+            std::tuple<int, uint64_t> count_bytes = j->second;
+            std::get<0>(count_bytes) += 1;
+            std::get<1>(count_bytes) += msg.m_raw_message_size;
```

Comment:
UPDATE: this thread has many half-baked ideas, I'm happy to tinker with the C++ at some point to understand better & offer a singular recommendation =P 

----

since the tuple is only used for these two values (count & bytes), `std::pair` seems like it'd offer a cleaner interface. call site would look like this:
```suggestion
            ++count_bytes.first;
            count_bytes.second += msg.m_raw_message_size;
```

Reply:
oooo, nifty c++ism that could be convenient here -`std::tie` can be used to "unpack" a `std::pair` 

[docs here](https://en.cppreference.com/w/cpp/utility/tuple/tie) - might also be useful to simplify the `operator<` code.

I have to tinker to better understand exactly what the interface would be, but my current understanding is that it could offer something like `std::tie(count, bytes) = j->second;` and then can just use `count` and `bytes` as the variables to increment

Reply:
I love it! I actually didn't know the difference between a pair and a tuple so as a first step, I went ahead and converted it `std::pair`. 

It's possible this chunk of code may go away or be refactored when I remove the original stats I added to the `getnettotals` RPC, so I'm going to revisit the `std::tie` suggestion in a little bit :) Interesting that you can name the vars with that data type!

Reply:
Can't promise I'll remember the differences between `std::tuple`, `std::pair`, and `std::tie` if this comes up again, but I'll try :) Resolving this comment now that the `count_bytes` data structure no longer exists.